fleet_notifications/state_checker.py

```diff
@@ -1,110 +1,152 @@
+        return max(self.orders, key=lambda order: order.timestamp, default=0)
+    
+
+    def _check_if_order_is_new(self, car_id: int, state: OrderState, admin_phone: str, under_test: bool) -> None:
+        """Adds the order belonging to the state to the list of orders. If the order is the first new one,
+        a notification is sent to the admin phone number. Returns false if the order can't be retrieved."""
+        no_active_order = True
+        for order in self.orders.values():
+            if order.car_id == car_id:
+                no_active_order = False
+                break
+        if state.order_id not in self.orders or state.status == OrderStatus.CANCELED:
+            try:
+                self.orders[state.order_id] = self.order_api.get_order(car_id=car_id, order_id=state.order_id)
+            except:
+                logger.error(f"Error while getting order with ID {state.order_id} from the api.")
+                return False
+        new_order_started = False
+        for order in self.orders.values():
+            if order.car_id == car_id:
+                new_order_started = True
+                break
+
+        if (no_active_order and new_order_started and self.orders[state.order_id].last_state.status
+            != OrderStatus.DONE and self.orders[state.order_id].last_state.status != OrderStatus.CANCELED):
+            logger.info(f"New mission started.")
+            threading.Thread(target=self.notification_client.call_phone, daemon=True,
+                             args=(admin_phone, under_test,)).start()
+        return True
+
+
+    def _check_if_order_is_done(self, car_id: int, state: OrderState, under_test: bool) -> None:
+        """Checks if the order is newly done and sends a notification to the phone number in the order."""
+        if self.orders[state.order_id].last_state.status != OrderStatus.DONE and state.status == OrderStatus.DONE:
+            logger.info(f"Order {state.order_id} is done.")
+            self.orders[state.order_id] = self.order_api.get_order(car_id=car_id, order_id=state.order_id)
+
+            notification_phone = self.orders[state.order_id].notification_phone
+            if(notification_phone is None):
+                logger.warning(f"Order {state.order_id} has no notification phone number.")
+                return
+
+            threading.Thread(target=self.notification_client.call_phone, daemon=True, args=(
+                             notification_phone.phone, under_test,)).start()
+
+
+    def _remove_finished_orders(self, since: int) -> None:
+        """Removes finished orders from the list and the database. If an order is not finished,
+        the since parameter is used as its latest timestamp."""
+        orders_to_remove = []
+        for order in self.orders.values():
+            if order.last_state.status == OrderStatus.DONE or order.last_state.status == OrderStatus.CANCELED:
+                orders_to_remove.append(order.id)
+            else:
```

Comment:
Split this into two methods, the first one cleaning up finished or deleted orders, and the other updating timestamp. 

```suggestion
    def _remove_finished_orders(self) -> None:
        """Removes finished orders from the list and the database."""
        finished_orders = [order for order in self.orders.values() if self._is_finished(order)]
        deleted_orders = [
            order for order in self.orders.values() if order not in self.order_api.get_orders()
        ]
        for order in finished_orders + deleted_orders:
            self.orders.pop(order.id)
            notifications_db.delete_order(order.id)

    def _update_latest_timestamps(self, since: int) -> None:
        """If an order is not finished, the since parameter is used as its latest timestamp."""
        orders_to_update = [order for order in self.orders.values() if not self._is_finished(order)]
        for order in orders_to_update:
            notifications_db.update_order(order.id, order.car_id, since)
```